src/pages/Login.js

```diff
@@ -10,69 +9,82 @@ class Login extends Component {
     this.state = {
       email: '',
       password: '',
-      redirect: false,
-      users: [],
+      flash: '',
+      user: {},
+      token: '',
     };
 
     this.handleInputChange = this.handleInputChange.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
+    this.setLocalStorage = this.setLocalStorage.bind(this);
+    this.notify = this.notify.bind(this);
   }
 
-  handleInputChange = event => {
-    // will update state by name attribute, so state name should be same with it
-    const {value, name} = event.target;
+  handleInputChange = e => {
     this.setState({
-      [name]: value,
+      [e.target.name]: e.target.value,
+      flash: '',
     });
   };
 
   onSubmit = event => {
-    event.preventDefault();
-    let userArray = this.state.users;
-    let existUser = userArray.find(data => {
-      return data.email === this.state.email; // if can't find, it will return undefined
-    });
-    if (existUser !== undefined) {
-      if (existUser.password === this.state.password) {
-        localStorage.setItem('user', JSON.stringify(existUser));
-        alert('Login successful');
-        this.setState({redirect: true});
-      } else {
-        alert('Login failed: password is wrong');
-        this.setState({password: ''});
-      }
-    } else {
-      alert('Login failed: you are not registered');
-      this.setState({email: '', password: ''});
+    if (event) {
+      event.preventDefault();
     }
-  };
 
-  componentDidMount() {
-    fetch(`${config.apiUrl}/users`)
+    const payload = {
+      email: this.state.email,
+      password: this.state.password,
+    };
+    fetch(`${config.apiUrl}/auth/login`, {
+      method: 'POST',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+      }),
+      body: JSON.stringify(payload),
+    })
       .then(res => res.json())
-      .then(data => {
-        this.setState({
-          users: data.users,
-        });
-      })
-      .catch(error => console.log(error));
+      .then(res =>
+        res.token
+          ? this.setState(
+              {
+                token: res.token,
+                flash: res.flash,
+                user: res.data,
+              },
+              () => this.setLocalStorage(),
+            )
+          : this.setState({
+              flash: res.flash,
+            }),
+      )
+      .catch(err => console.log(err));
+  };
+
+  setLocalStorage() {
+    localStorage.setItem('token', this.state.token);
+    localStorage.setItem('user', JSON.stringify(this.state.user));
+
+    this.notify();
+  }
+
+  notify() {
+    var event = new Event('authenticated');
+    document.dispatchEvent(event);
```

Comment:
what does this do?

Reply:
It creates new Event 'authenticated' and says to the document that event 'authenticated' happened. In simple words: sending event/subscribing to event allows completely unrelated components to talk to each other in case there's no Redux setup. But with Redux if you refresh the page Redux state in store disappears, so I didn't find another solution.

https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent